addons/contexts/src/manager/components/ToolBar.test.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { ToolBar } from './ToolBar';
+
+describe('Tests on addon-contexts component: ToolBar', () => {
+  it('should render nothing if receive an empty contextNodes', () => {
+    // when
+    const result = shallow(<ToolBar nodes={[]} state={{}} setSelected={jest.fn} />);
```

Comment:
Are these better expressed as storybook stories & tested with storyshots?

Reply:
@shilman, I was thinking about this previously, and even try to use `react-testing-library` at very beginning... But later I found if I really want to test them, I have to find its ThemeProvider first; otherwise, I will get some object node missed and React get crashed.

That is why I roll it back to more traditional shallow testing approach (to stay inside the addon itself). Right now everything is in the monorepo and user will have to use `@storybook/theme` implicitly so I guess it should be fine?!  But I somehow think the `@storybook/components` should have some components out-of-the-box for the addon creator.

To be more specific on the scope here, the only thing really need to get tested is the `ToolBarControl` component.  This component is responsible for validating the selected name (and some states as you may see there). I added other testing for parity.

There are some part I want to test but not get it tested. Anyway I will need to rethink about testing here. Thanks for suggestion though 🙏.

Reply:
Check out my addon improvements PR @leoyli 

https://github.com/storybooks/storybook/pull/6759

It adds a component for addon creation, If you have idea on what components would be useful for addon creators, let's chat about that.

Reply:
Woo nice, I will take a look, I will also migrate `@storybook/components` first, and reach you out 👍